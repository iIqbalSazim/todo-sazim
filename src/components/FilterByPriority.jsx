import { Button } from "@mantine/core";
import { IconX } from "@tabler/icons-react";
import { PRIORITY } from "../constants/constant";
import { COLORS } from "../constants/constant";

const FilterByPriority = ({ setFilter, filter }) => {
  return (
    <>
      <Button.Group>
        <Button
          variant={
            filter.priority === PRIORITY.HIGH && !filter.dueDate
              ? "filled"
              : "outline"
          }
          color={COLORS.PRIORITYBUTTON}
          onClick={() => setFilter(PRIORITY.HIGH)}
        >
          High
        </Button>
        <Button
          variant={
            filter.priority === PRIORITY.MEDIUM && !filter.dueDate
              ? "filled"
              : "outline"
          }
          color={COLORS.PRIORITYBUTTON}
          onClick={() => setFilter(PRIORITY.MEDIUM)}
        >
          Medium
        </Button>
        <Button
          variant={
            filter.priority === PRIORITY.LOW && !filter.dueDate
              ? "filled"
              : "outline"
          }
          color={COLORS.PRIORITYBUTTON}
          onClick={() => setFilter(PRIORITY.LOW)}
        >
          Low
        </Button>
        <Button
          variant="light"
          color={"indigo.4"}
          size="sm"
          rightIcon={<IconX size={"1rem"} />}
          onClick={() => setFilter("")}
        >
          Clear Filter
        </Button>
      </Button.Group>
    </>
  );
};

export default FilterByPriority;
