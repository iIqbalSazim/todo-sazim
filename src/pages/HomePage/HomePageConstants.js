export const COLORS = {
  BUTTON: "indigo.7",
  PRIORITY_BUTTON: "indigo.3",
  DISPLAY_DATE: "gray.8",
};

export const COMPLETED_STATUS = {
  ALL: "all",
  ACTIVE: "active",
  COMPLETED: "completed",
};

export const PRIORITY = {
  HIGH: "high",
  MEDIUM: "medium",
  LOW: "low",
};

export const PRIORITY_OPTIONS = [
  { value: "high", label: "High" },
  { value: "medium", label: "Medium" },
  { value: "low", label: "Low" },
];
