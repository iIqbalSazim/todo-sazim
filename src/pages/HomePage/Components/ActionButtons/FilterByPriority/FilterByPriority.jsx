import { Button, MediaQuery } from "@mantine/core";
import { IconX } from "@tabler/icons-react";

import { PRIORITY, COLORS } from "../../../HomePageConstants";

const FilterByPriority = ({ setFilter, filter }) => {
  return (
    <>
      <MediaQuery smallerThan={"md"} styles={{ display: "none" }}>
        <Button.Group>
          <Button
            variant={
              filter.priority === PRIORITY.HIGH && !filter.dueDate
                ? "filled"
                : "outline"
            }
            color={COLORS.PRIORITY_BUTTON}
            onClick={() => setFilter(PRIORITY.HIGH)}
            data-testid="high-button-sm"
          >
            High
          </Button>
          <Button
            variant={
              filter.priority === PRIORITY.MEDIUM && !filter.dueDate
                ? "filled"
                : "outline"
            }
            color={COLORS.PRIORITY_BUTTON}
            onClick={() => setFilter(PRIORITY.MEDIUM)}
            data-testid="medium-button-sm"
          >
            Medium
          </Button>
          <Button
            variant={
              filter.priority === PRIORITY.LOW && !filter.dueDate
                ? "filled"
                : "outline"
            }
            color={COLORS.PRIORITY_BUTTON}
            onClick={() => setFilter(PRIORITY.LOW)}
            data-testid="low-button-sm"
          >
            Low
          </Button>
          <Button
            variant="light"
            color={"indigo.4"}
            size="sm"
            rightIcon={<IconX size={"1rem"} />}
            onClick={() => setFilter("")}
          >
            Clear Filter
          </Button>
        </Button.Group>
      </MediaQuery>

      <MediaQuery largerThan={"md"} styles={{ display: "none" }}>
        <Button.Group>
          <Button
            variant={
              filter.priority === PRIORITY.HIGH && !filter.dueDate
                ? "filled"
                : "outline"
            }
            color={COLORS.PRIORITYBUTTON}
            onClick={() => setFilter(PRIORITY.HIGH)}
            size="xs"
            data-testid="high-button-md"
          >
            High
          </Button>
          <Button
            variant={
              filter.priority === PRIORITY.MEDIUM && !filter.dueDate
                ? "filled"
                : "outline"
            }
            color={COLORS.PRIORITYBUTTON}
            onClick={() => setFilter(PRIORITY.MEDIUM)}
            size="xs"
            data-testid="medium-button-md"
          >
            Medium
          </Button>
          <Button
            variant={
              filter.priority === PRIORITY.LOW && !filter.dueDate
                ? "filled"
                : "outline"
            }
            color={COLORS.PRIORITYBUTTON}
            onClick={() => setFilter(PRIORITY.LOW)}
            size="xs"
            data-testid="low-button-md"
          >
            Low
          </Button>
          <Button
            variant="light"
            color={"indigo.4"}
            size="xs"
            rightIcon={<IconX size={"1rem"} />}
            onClick={() => setFilter("")}
          >
            Clear
          </Button>
        </Button.Group>
      </MediaQuery>
    </>
  );
};

export default FilterByPriority;
