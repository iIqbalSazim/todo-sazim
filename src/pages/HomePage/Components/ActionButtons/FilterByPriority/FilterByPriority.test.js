import { screen, render } from "@testing-library/react";
import "@testing-library/jest-dom";

import FilterByPriority from "./FilterByPriority";
import userEvent from "@testing-library/user-event";
import { PRIORITY } from "../../../HomePageConstants";

const mockSetFilter = jest.fn();

describe("FilterByPriority", () => {
  test("renders correctly", () => {
    const { container } = render(
      <FilterByPriority setFilter={mockSetFilter} filter={{}} />
    );
    expect(container).toMatchSnapshot();

    const highButtons = screen.getAllByText("High");
    const mediumButtons = screen.getAllByText("Medium");
    const lowButtons = screen.getAllByText("Low");
    const clearFilterButton = screen.getByText("Clear Filter");
    const clearButton = screen.getByText("Clear Filter");

    expect(highButtons.length).toBe(2);
    expect(lowButtons.length).toBe(2);
    expect(mediumButtons.length).toBe(2);
    expect(clearFilterButton).toBeInTheDocument();
    expect(clearButton).toBeInTheDocument();
  });

  test("activates the high filter when clicked on 'High' button", async () => {
    render(<FilterByPriority setFilter={mockSetFilter} filter={{}} />);

    const highButtonMd = screen.getByTestId("high-button-md");
    await userEvent.click(highButtonMd);
    expect(mockSetFilter).toHaveBeenCalledWith(PRIORITY.HIGH);

    const highButtonSm = screen.getByTestId("high-button-sm");
    await userEvent.click(highButtonSm);
    expect(mockSetFilter).toHaveBeenCalledWith(PRIORITY.HIGH);
  });

  test("activates the medium filter when clicked on 'Medium' button", async () => {
    render(<FilterByPriority setFilter={mockSetFilter} filter={{}} />);

    const mediumButtonMd = screen.getByTestId("medium-button-md");
    await userEvent.click(mediumButtonMd);
    expect(mockSetFilter).toHaveBeenCalledWith(PRIORITY.MEDIUM);

    const mediumButtonSm = screen.getByTestId("medium-button-sm");
    await userEvent.click(mediumButtonSm);
    expect(mockSetFilter).toHaveBeenCalledWith(PRIORITY.MEDIUM);
  });

  test("activates the low filter when clicked on 'Low' button", async () => {
    render(<FilterByPriority setFilter={mockSetFilter} filter={{}} />);

    const lowButtonMd = screen.getByTestId("low-button-md");
    await userEvent.click(lowButtonMd);
    expect(mockSetFilter).toHaveBeenCalledWith(PRIORITY.LOW);

    const lowButtonSm = screen.getByTestId("low-button-sm");
    await userEvent.click(lowButtonSm);
    expect(mockSetFilter).toHaveBeenCalledWith(PRIORITY.LOW);
  });

  test("clears filter when clicked on 'Clear Filter' or 'Clear' button", async () => {
    render(<FilterByPriority setFilter={mockSetFilter} filter={{}} />);

    const clearFilterButton = screen.getByRole("button", {
      name: "Clear Filter",
    });
    await userEvent.click(clearFilterButton);
    expect(mockSetFilter).toHaveBeenCalledWith("");

    const clearButton = screen.getByRole("button", {
      name: "Clear",
    });
    await userEvent.click(clearButton);
    expect(mockSetFilter).toHaveBeenCalledWith("");
  });
});
