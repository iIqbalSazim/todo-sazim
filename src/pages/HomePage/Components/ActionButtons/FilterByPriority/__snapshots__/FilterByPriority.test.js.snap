// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterByPriority renders correctly 1`] = `
<div>
  <div
    class="mantine-ButtonGroup-root mantine-MediaQuery-media mantine-1s7mmb2"
  >
    <button
      class="mantine-UnstyledButton-root mantine-Button-root mantine-ovgvo7"
      data-button="true"
      data-testid="high-button-sm"
      type="button"
    >
      <div
        class="mantine-1wpc1xj mantine-Button-inner"
      >
        <span
          class="mantine-1ryt1ht mantine-Button-label"
        >
          High
        </span>
      </div>
    </button>
    <button
      class="mantine-UnstyledButton-root mantine-Button-root mantine-ovgvo7"
      data-button="true"
      data-testid="medium-button-sm"
      type="button"
    >
      <div
        class="mantine-1wpc1xj mantine-Button-inner"
      >
        <span
          class="mantine-1ryt1ht mantine-Button-label"
        >
          Medium
        </span>
      </div>
    </button>
    <button
      class="mantine-UnstyledButton-root mantine-Button-root mantine-ovgvo7"
      data-button="true"
      data-testid="low-button-sm"
      type="button"
    >
      <div
        class="mantine-1wpc1xj mantine-Button-inner"
      >
        <span
          class="mantine-1ryt1ht mantine-Button-label"
        >
          Low
        </span>
      </div>
    </button>
    <button
      class="mantine-UnstyledButton-root mantine-Button-root mantine-ubwojs"
      data-button="true"
      type="button"
    >
      <div
        class="mantine-1wpc1xj mantine-Button-inner"
      >
        <span
          class="mantine-1ryt1ht mantine-Button-label"
        >
          Clear Filter
        </span>
        <span
          class="mantine-Button-icon mantine-Button-rightIcon mantine-467wo8"
        >
          <svg
            class="tabler-icon tabler-icon-x"
            fill="none"
            height="1rem"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 6l-12 12"
            />
            <path
              d="M6 6l12 12"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    class="mantine-ButtonGroup-root mantine-MediaQuery-media mantine-1fyndrs"
  >
    <button
      class="mantine-UnstyledButton-root mantine-Button-root mantine-8hn0e"
      data-button="true"
      data-testid="high-button-md"
      type="button"
    >
      <div
        class="mantine-1wpc1xj mantine-Button-inner"
      >
        <span
          class="mantine-1ryt1ht mantine-Button-label"
        >
          High
        </span>
      </div>
    </button>
    <button
      class="mantine-UnstyledButton-root mantine-Button-root mantine-8hn0e"
      data-button="true"
      data-testid="medium-button-md"
      type="button"
    >
      <div
        class="mantine-1wpc1xj mantine-Button-inner"
      >
        <span
          class="mantine-1ryt1ht mantine-Button-label"
        >
          Medium
        </span>
      </div>
    </button>
    <button
      class="mantine-UnstyledButton-root mantine-Button-root mantine-8hn0e"
      data-button="true"
      data-testid="low-button-md"
      type="button"
    >
      <div
        class="mantine-1wpc1xj mantine-Button-inner"
      >
        <span
          class="mantine-1ryt1ht mantine-Button-label"
        >
          Low
        </span>
      </div>
    </button>
    <button
      class="mantine-UnstyledButton-root mantine-Button-root mantine-11592gy"
      data-button="true"
      type="button"
    >
      <div
        class="mantine-1wpc1xj mantine-Button-inner"
      >
        <span
          class="mantine-1ryt1ht mantine-Button-label"
        >
          Clear
        </span>
        <span
          class="mantine-Button-icon mantine-Button-rightIcon mantine-467wo8"
        >
          <svg
            class="tabler-icon tabler-icon-x"
            fill="none"
            height="1rem"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 6l-12 12"
            />
            <path
              d="M6 6l12 12"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
</div>
`;
